geometry/include/pcl/geometry/mesh_indices.h

```diff
@@ -38,48 +38,49 @@
  *
  */
 
-// NOTE: This file has been created with
-// 'pcl_src/geometry/include/pcl/geometry/mesh_indices.py'
-
 #pragma once
 
 #include <pcl/geometry/boost.h>
 
 #include <iostream>
 
 ////////////////////////////////////////////////////////////////////////////////
-// VertexIndex
+// MeshIndex
 ////////////////////////////////////////////////////////////////////////////////
 
 namespace pcl {
-namespace geometry {
-/** \brief Index used to access elements in the half-edge mesh. It is basically just a
- * wrapper around an integer with a few added methods.
- * \author Martin Saelzle
- * \ingroup geometry
- */
-class VertexIndex
+namespace detail {
+
+template <class IndexTagT>
+class MeshIndex;
+
+template <class IndexTagT>
+std::istream&
+operator>>(std::istream& is, MeshIndex<IndexTagT>&);
```

Comment:
Why are these lines needed? They weren't necessary before, and the class/operator are defined immediately afterwards.

Reply:
The issue was due to the friend operator overload with a templated argument. The solution to pre-declare the templated operator prevents to have all template instantiations of the class to be friend with all the templated instantiations of the operator ([Johannes Schaub answer](https://stackoverflow.com/questions/1297609/overloading-friend-operator-for-template-class)). A viable alternative is to define it directly into the class.
However, I decided to stick with this solution to keep the structure of the operators from the previous classes, for example I did not declared operator << to be friend, thus for it this issue is not present.

Tell me if you want me to change it, then I will rebase asap!

Reply:
Thanks for the explanation! Then please just rebase onto master to make the CI pass.

Reply:
You are welcome! 
Since I am here, when I rebase to this branch, it says that my branches diverged (which is normal). It is ok to just force push those commits as I would do normally?
I don't know how rebase affects an open PR so just to be sure to do not mess it 😆

Reply:
As dangerous as a force push can be in some other situations, here it is ok. The open PR will be fine :smile:

Reply:
Usually, don't force-push if the PR is in the middle of a review (because GitHub loses count of which files have been marked as read on a force-push)

Once the review is complete and everyone is happy (or if the review hasn't started), you can force-push safely (unless you are collaborating with someone on that branch)

I have a git alias that helps me force-push safely during collaborations:
```
[alias]
  force-push = push --force-with-lease
```

Reply:
Rebased smoothly, thank you guys!